app/controllers/discourse_ai/embeddings/embeddings_controller.rb

```diff
@@ -19,12 +18,15 @@ def search
             use_pg_headlines_for_excerpt: false,
           )
 
-        DiscourseAi::Embeddings::SemanticSearch
-          .new(guardian)
-          .search_for_topics(query, page)
-          .each { |topic_post| grouped_results.add(topic_post) }
+        semantic_search = DiscourseAi::Embeddings::SemanticSearch.new(guardian)
```

Comment:
Can we introduce a rate limit here? A shared one for all anons and a per user for logged in?